### ISSUE TITLE
[White Label] Add configuration into a single model/table instead of having many columns

### ISSUE THREAD

**Issue by jibees** [openfoodfoundation/openfoodnetwork#10749]
### What we should change and why (this is tech debt)
This is an open discussion but adding many attribute/columns to enterprise model/table doesn't seems fine to me: adding columns to a so used table isn't good for performance. 
That's why I propose to add model/table for white labeling configuration instead of having severable attributes.

Table name: `WhiteLableConfiguration`
columns: 
 - `enterprise_id`: FK to enterprise.id
 - `hide_ofn_navigation, :boolean, null: false, default: false`
 - .... 

### Context
This came up when developing White labelling.

### Impact and timeline

 - This needs data migration. 
 - This needs to be done once has been merged: https://github.com/openfoodfoundation/openfoodnetwork/pull/10744 (or requires 10744 to be rebase onto this PR)

**Comment by mkllnk** [comment]
I agree. A separate model for the configuration is the best solution.
